packages/create-fuels/test/cli.test.ts

```diff
@@ -16,11 +19,20 @@ import { generateArgv } from './utils/generateArgs';
 import { mockLogger } from './utils/mockLogger';
 import { filterOriginalTemplateFiles, getAllFiles } from './utils/templateFiles';
 
+vi.mock('../src/lib/doesTemplateExist', async () => {
```

Comment:
Did you try `spyOn` this method?

Nitty but I quite like the [`mockAll`](https://github.com/FuelLabs/fuels-ts/blob/6eb24007258ed946c177369cba648b5bd6ad2535/packages/fuels/src/cli/commands/deploy/index.test.ts#L16-L29) function in tests that we seem to be adhering to.

Reply:
@petertonysmith94 `spyOn` only lets you _inspect_ a function and not actually mock it if I'm not wrong. In this case, I wanted to actually mock the implementation/return values

Reply:
The `vi.mock` call is hoisted, and in most cases that's not what you want.

https://github.com/FuelLabs/fuels-ts/blob/fe06f0de2ebc4cdbc83614b085568db1f0b45eae/packages/create-fuels/src/lib/tryInstallFuelUp.test.ts#L36-L38

Reply:
Oh I wasn't aware of the hoisting @arboleya, thanks for the heads-up. In that case, maybe we should use `vi.fn().mockImplementation` then

According to Vitest docs, `spyOn` will not let you change the behaviour of any functions: https://vitest.dev/guide/mocking.html#functions 

I remember that in a recent PR, I spotted that some of our mock implementations were using this pattern and as a result of that, the functions weren't being mocked at all. This might be the case for the example you have quoted as well

Reply:
To mock, you need to call `spyOn` followed by `mockImplementation`.

Reply:
Done. Apologies for the confusion

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2851/commits/cf0ab4c845f88df3dad60db2ce3eb2a3f1f3071f